File: functions/src/dto/friend_request.ts
export interface FriendRequest {
    id: string
    sender: FriendRequestProfile
    receiver: FriendRequestProfile
    status: string
}

export interface FriendRequestProfile {
    id: string
    username: string
    displayName?: string
    profilePictureIndex: number
}

export enum FriendRequestStatus {
    "PENDING", "DECLINED", "ACCEPTED", "CANCELED"
}

File: composeApp/src/commonMain/kotlin/com/makapp/kaizen/domain/models/friend/FriendRequest.kt
package com.makapp.kaizen.domain.models.friend

import kotlinx.serialization.Serializable

@Serializable
data class FriendRequest(
    val id: String,
    val sender: FriendRequestProfile,
    val receiver: FriendRequestProfile,
    val status: Status
) {
    @Serializable
    enum class Status {
        PENDING, DECLINED, ACCEPTED, CANCELED
    }
}

@Serializable
data class FriendRequestProfile(
    val id: String,
    val username: String,
    val displayName: String? = null,
    val profilePictureIndex: Int? = null
) {
    fun getName() = if (!displayName.isNullOrBlank())
        displayName
    else
        username
}
